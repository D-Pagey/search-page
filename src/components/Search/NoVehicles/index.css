.no-vehicles-container {
    background-color: white;
    border: 1px solid #e4e3e3;
    border-radius: 6px;
    padding: 16px;
    width: 290px;
}

.no-vehicles-title {
    font-size: 28px;
    letter-spacing: -1.2px;
    line-height: 1.1;
    margin: 0;
    text-align: center;
    text-transform: uppercase;
}

.no-vehicles-form {
    display: flex;
    flex-direction: column;
}

.form-group {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;
}

.no-vehicles-label {
    color: black;
    font-size: 16px;
    font-weight: bold;
    margin-bottom: 0.2rem;
}

.form-input {
    height: 40px;
    padding: 0 10px;
    border: 1px solid #e4e3e3;
    border-radius: 6px;
    font-size: 1rem;
}

.row {
    flex-direction: row;
}

.checkbox { 
    height: 50px;
    width: 50px;
}

.updates-copy {
    padding-left: 20px;
}