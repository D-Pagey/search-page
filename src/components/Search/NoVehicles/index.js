import React from 'react';

import './index.css';
import Button from '../../Button';

export default function NoVehicles() {
  return (
    <div className="no-vehicles-container">
      <div>
        <h3 className="no-vehicles-title">looking for a specific car?</h3>
        <p>Let us know what car you are looking for, and we&apos;ll be in touch!</p>
      </div>
      <form className="no-vehicles-form">
        <div className="form-group">
          <label htmlFor="fullname" className="no-vehicles-label">Full Name</label>
          <input
            type="text"
            required
            id="fullname"
            name="fullname"
            className="form-input"
            placeholder="John Doe"
          />
        </div>
        <div className="form-group">
          <label htmlFor="email" className="no-vehicles-label">Email address</label>
          <input
            type="email"
            required
            id="email"
            name="email"
            className="form-input"
            placeholder="example@gmail.com"
          />
        </div>
        <div className="form-group">
          <label htmlFor="time-frame" className="no-vehicles-label">
          When do you want a car?
          </label>
          <select
            type="text"
            required
            id="time-frame"
            name="time-frame"
            className="form-input"
            placeholder="example@gmail.com"
          >
            <option value="this-month">This Month</option>
            <option value="3-months">Within 3 Months</option>
            <option value="6-months">Within 6 Months</option>
            <option value="more-6-months">6 Months +</option>
          </select>
        </div>
        <div className="form-group">
          <label htmlFor="notes" className="no-vehicles-label">
          What are you looking for?
          </label>
          <input
            type="text"
            id="notes"
            name="notes"
            className="form-input"
            placeholder="A White BMW M3"
          />
        </div>
        <div className="form-group row">
          <input
            type="checkbox"
            id="checkbox"
            name="checkbox"
            className="checkbox"
          />
          <label htmlFor="checkbox" className="updates-copy">I would like to
          receive updates about discounts, special offers, news and events from Drover.
          </label>
        </div>
        <Button text="Find my car" />
      </form>
    </div>
  );
}
