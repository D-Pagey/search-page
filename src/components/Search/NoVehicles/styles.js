import styled from 'styled-components';

export const NoVehiclesWrapper = styled.div`
  background-color: white;
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  margin-bottom: 40px;
  padding: 16px;
  @media (min-width: 768px) {
    width: 690px;
  }
`;

export const NoVehiclesHeader = styled.div`
  @media (min-width: 768px) {
    margin: 0 auto;
    width: 98%;
  }
`;

export const NoVehiclesTitle = styled.h3`
  font-size: 28px;
  letter-spacing: -1.2px;
  line-height: 1.1;
  margin: 0;
  text-align: center;
  text-transform: uppercase;
`;

export const NoVehiclesForm = styled.form`
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
`;

export const NoVehiclesLabel = styled.label`
  color: black;
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 0.2rem;
`;

export const UpdatesCopy = styled.p`
  padding-left: 20px;
`;

export const FormInput = styled.input`
  height: 40px;
  padding: 0 10px;
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  font-size: 1rem;
`;

export const FormSelect = styled.select`
  height: 40px;
  padding: 0 10px;
  border: 1px solid #e4e3e3;
  border-radius: 6px;
  font-size: 1rem;
`;

/* eslint-disable */
export const FormGroup = styled.div`
  display: flex;
  flex-direction: ${props => props.row ? 'row' : 'column'};
  margin-bottom: 10px;
  @media (min-width: 768px) {
    &:nth-of-type(2) {
        margin: 0 10px;
    }

    &:nth-of-type(4) {
        width: 98%;
    }
  }
`;
/* eslint-enable */
