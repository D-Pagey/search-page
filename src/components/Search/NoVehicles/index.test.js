import React from 'react';
import { shallow } from 'enzyme';
import NoVehicles from '.';

it('should render', () => {
  const wrapper = shallow(<NoVehicles />);
  expect(wrapper).toMatchSnapshot();
});
