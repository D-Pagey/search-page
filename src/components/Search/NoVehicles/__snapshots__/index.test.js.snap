// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="no-vehicles-container"
>
  <div>
    <h3
      className="no-vehicles-title"
    >
      looking for a specific car?
    </h3>
    <p>
      Let us know what car you are looking for, and we'll be in touch!
    </p>
  </div>
  <form
    className="no-vehicles-form"
  >
    <div
      className="form-group"
    >
      <label
        className="no-vehicles-label"
        htmlFor="fullname"
      >
        Full Name
      </label>
      <input
        className="form-input"
        id="fullname"
        name="fullname"
        placeholder="John Doe"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="no-vehicles-label"
        htmlFor="email"
      >
        Email address
      </label>
      <input
        className="form-input"
        id="email"
        name="email"
        placeholder="example@gmail.com"
        required={true}
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="no-vehicles-label"
        htmlFor="time-frame"
      >
        When do you want a car?
      </label>
      <select
        className="form-input"
        id="time-frame"
        name="time-frame"
        placeholder="example@gmail.com"
        required={true}
        type="text"
      >
        <option
          value="this-month"
        >
          This Month
        </option>
        <option
          value="3-months"
        >
          Within 3 Months
        </option>
        <option
          value="6-months"
        >
          Within 6 Months
        </option>
        <option
          value="more-6-months"
        >
          6 Months +
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <label
        className="no-vehicles-label"
        htmlFor="notes"
      >
        What are you looking for?
      </label>
      <input
        className="form-input"
        id="notes"
        name="notes"
        placeholder="A White BMW M3"
        type="text"
      />
    </div>
    <div
      className="form-group row"
    >
      <input
        className="checkbox"
        id="checkbox"
        name="checkbox"
        type="checkbox"
      />
      <label
        className="updates-copy"
        htmlFor="checkbox"
      >
        I would like to receive updates about discounts, special offers, news and events from Drover.
      </label>
    </div>
    <Button
      onClick={null}
    >
      Find my car
    </Button>
  </form>
</div>
`;
